oci/oci_image.go

```diff
@@ -0,0 +1,278 @@
+package oci
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"time"
+
+	"github.com/nanovms/ops/lepton"
+	"github.com/nanovms/ops/types"
+	"github.com/olekukonko/tablewriter"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-go-sdk/objectstorage"
+	"github.com/oracle/oci-go-sdk/workrequests"
+	"github.com/schollz/progressbar/v3"
+)
+
+var (
+	qcow2ImagesDir = lepton.GetOpsHome() + "/qcow2-images"
+)
+
+// BuildImage creates local image
+func (p *Provider) BuildImage(ctx *lepton.Context) (string, error) {
+	c := ctx.Config()
+	err := lepton.BuildImage(*c)
+	if err != nil {
+		return "", err
+	}
+
+	imagePath, err := p.createQcow2Image(c)
+	if err != nil {
+		return "", err
+	}
+
+	err = os.Remove(c.RunConfig.Imagename)
+	if err != nil {
+		return "", err
+	}
+
+	return imagePath, nil
+}
+
+func (p *Provider) createQcow2Image(c *types.Config) (imagePath string, err error) {
+	qcow2ImagesDir, err := findOrCreateqcow2ImagesDir()
+	if err != nil {
+		return
```

Comment:
return err?

Reply:
Yes. If we initialize the variables on return arguments (`(imagePath string, err error)`) we don't have to specify values on return.

Reply:
so yes. The error is returned.

Reply:
ah i see it's named ok